spinalcordtoolbox/deepseg/nnunet.py

```diff
@@ -0,0 +1,34 @@
+import os
+import torch
+
+from batchgenerators.utilities.file_and_folder_operations import join
+from nnunetv2.inference.predict_from_raw_data import nnUNetPredictor
+
+
+def create_nnunet_from_plans(path_model):
+    use_best_checkpoint = False
+    tile_step_size = 0.5
+    folds_avail = [int(f.split('_')[-1]) for f in os.listdir(path_model) if f.startswith('fold_')]
```

Comment:
This should be changed to 
```suggestion
folds_avail = 'all' if os.listdir(path_model) == 'fold_all' else [int(f.split('_')[-1]) for f in os.listdir(path_model) if f.startswith('fold_')]
```
We recently realized that there could also be `fold_all` so we updated the code in the repo to accommodate this as well. tagging @valosekj

Reply:
> Now, if a user wants to run prediction using multiple folds, they have to explicitly specify them, e.g., `-fold 2,3`. If `-fold all` is used, then the inference is run using `fold_all`. Hope the explanation is clear.

When you say "the user", do you mean the user of `sct_deepseg`? Or do you mean "the user of nnunet" (i.e. us developers), and thus we would keep this hidden from SCT's users?

Reply:
Good question! Actually, I can see that `sct_deepseg` has no input arg "-fold," meaning that it is not possible to choose between folds, right? And moreover, IIRC, the final models are `fold_1` for SCIseg and `fold_all` rootlets. In other words, we share only a single fold for each model, and there is thus no point in selecting other folds since they will not be available. So the line can be rewritten to simply take the only available fold (`fold_1` for SCIseg and `fold_all` rootlets).

Reply:
> So the line can be rewritten to simply take the only available fold (fold_1 for SCIseg and fold_all rootlets).

Instead of specifiying `fold_1`, we can keep the line 

```
folds_avail = 'all' if os.listdir(path_model) == 'fold_all' else [int(f.split('_')[-1]) for f in os.listdir(path_model) if f.startswith('fold_')]
```

Reason is that the `fold_x` might be different for different models (i.e. the ones that would come in the future). Makes the code a bit more modular (?), imo

Reply:
Agree! This is exactly what I meant!

Reply:
I applied the suggestion, though it looks as though the tests still failed with the newest rootlet model:

> ```
>  ValueError: invalid literal for int() with base 10: 'all'
> ```

The issue is pretty simple! `os.listdir()` returns more than just the model folder, heh:

![image](https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/16181459/ffc430f8-a4a1-478a-ba6c-74d48355e89d)

This should be easy to fix. :)